services/src/main/java/org/fao/geonet/api/ApiUtils.java

```diff
@@ -119,13 +119,15 @@ public static String getInternalId(String uuidOrInternalId, Boolean approved)
             id = String.valueOf(metadata.getId());
         }
 
-        if (StringUtils.isEmpty(id)) {
+        if (!StringUtils.hasLength(id) && Lib.type.isInteger(id)) {
```

Comment:
`Lib.type.isInteger(id)` doesn't seem necessary, no? `!StringUtils.hasLength(id)` checks for empty value

Reply:
metadataUtils.findOne(uuidOrInternalId) fails if it is not a number.

![image](https://github.com/geonetwork/core-geonetwork/assets/1868233/d06b648d-a078-4903-92d4-7f28e9a04b7f)

Probably because of the following code.
![image](https://github.com/geonetwork/core-geonetwork/assets/1868233/1bef6b06-d019-4d36-89e4-33811977652d)

Reply:
@ianwallen maybe I'm confused, but if `!StringUtils.hasLength(id)` is `true` means that `id` is empty and should not enter in the `if` block, so the other check doesn't seem required?

Another stuff would be if instead it would be checking a non-empty value with `StringUtils.hasLength(id)`, then the 2on condition could make sense, no?

Reply:
You are correct - I now remember what happened.

I had initially used `StringUtils.hasLength(id)` instead of `!StringUtils.hasLength(id)` and got the error.

I pushed a fix.

TGIF :)

Reply:
Sorry - the issue was that I was checking the wrong value.  I was checking `id` when I should have check `uuidOrInternalId`

I also found another error where the following

http://127.0.0.1:8080/geonetwork/srv/api/records/000/status

was returning 400 when it should have return 404

Reply:
@ianwallen I still see the original code:

https://github.com/geonetwork/core-geonetwork/pull/8152/files#diff-be55588828d1d1d392e9a8e6cf8afcb9b012d9732266c496b8940e85f8126891R123

Reply:
@josegar74 
My original PR had the following 
`if (!StringUtils.hasLength(id) && Lib.type.isInteger(id)) {`
Now it is the following
`if (!StringUtils.hasLength(id) && Lib.type.isInteger(uuidOrInternalId)) {`

This is required because `metadataUtils.findOne(uuidOrInternalId)` assumes the value to be a number.
Without this fix, If you attempt to use uuid value of `"Test"`, it will not find the record on the first attempt and then it will attempt to call `metadataUtils.findOne("Test") ` and throw a numeric Exception.

So instead of returning an HTTP code of 404 because it could not find "Test" (or any invalid uuid), it would return a 400  because of the numeric exception.  By ensuring it is a number, it will return the correct 404.